packages/repository-json-schema/src/build-schema.ts

```diff
@@ -436,7 +463,20 @@ export function modelToJsonSchema<T extends object>(
 
     const propSchema = getJsonSchema(referenceType, options);
 
-    includeReferencedSchema(referenceType.name, propSchema);
+    // JSONSchema6Definition allows both boolean and JSONSchema6 types
+    if (typeof result.properties[p] !== 'boolean') {
+      const prop = result.properties[p] as JSONSchema;
+      const propTitle = propSchema.title ?? referenceType.name;
+      const targetRef = {$ref: `#/definitions/${propTitle}`};
```

Comment:
@raymondfeng We have a function to generate the title suffix in [here](https://github.com/strongloop/loopback-next/blob/6a41d752f7f7dbc9f85ad8d74eb38db71e65e7c7/packages/repository-json-schema/src/build-schema.ts#L331), and it handles the `partial` option, why do we check again here?

Reply:
The purpose is to make sure `ParentPartial` uses `ChildPartial` instead of `Child`.

Reply:
@jannyHou , if you're ok with @raymondfeng 's answer, please approve. thx.

Reply:
> The purpose is to make sure ParentPartial uses ChildPartial instead of Child.

Fair enough 👍